gadget/validate.go

```diff
@@ -35,6 +35,48 @@ type validationState struct {
 	SystemSave *VolumeStructure
 }
 
+// ValidationConstraints carries extra constraints on top of those
+// implied by the model to use for gadget validation.
+// They might be constraints that are determined only at runtime.
+type ValidationConstraints struct {
+	// EncryptedData when true indicates that the gadget will be used on a
+	// device where the data partition will be encrypted.
+	EncryptedData bool
+}
+
+// Validate validates the given gadget metadata against the consistency rules
+// for role usage, labels etc as implied by the model and extra constraints
+// that might be known only at runtime.
```

Comment:
Not really a problem, but _Validate validates_ sounds a bit unhelpful. Maybe (as below):`Validate checks that the given gadget metadata matches the consistency rules for role usage, labels, etc., as implied...`
`

Reply:
thanks, I'm applying this and the missing on in the follow-up